Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -1,16 +1,312 @@
+                    {
+                        Name = tagDto.Name,
+                    }).ToList();
+
+                var tagsToRemove = existingEvent.Tags
+                    .Where(existingTag => newTags.All(newTag => newTag.Name != existingTag.Name))
+                    .ToList();
+
+                foreach (var tagToRemove in tagsToRemove)
+                {
+                    existingEvent.Tags.Remove(tagToRemove);
+                }
+
+                var existingTagsInDb = context.Tags.ToList();
+                var tagsToAdd = newTags
+                    .Where(newTag => existingTagsInDb.All(existingTag => existingTag.Name != newTag.Name))
+                    .ToList();
+
+                context.Tags.AddRange(tagsToAdd);
+                context.SaveChanges();
+
+                var updatedTags = newTags
+                    .Select(newTag => existingTagsInDb.FirstOrDefault(existingTag => existingTag.Name == newTag.Name) ??
+                                      newTag)
+                    .ToList();
+
+                existingEvent.Tags = updatedTags;
+
+                context.SaveChanges();
+            }
+
+            if (eventDto.Parameters is not null)
+            {
+                var newParameters = eventDto.Parameters
+                    .Select(paramDto => new Parameter
+                    {
+                        Key = paramDto.Key,
+                        Value = paramDto.Value 
+                    }).ToList();
+
+                var paramsToRemove = existingEvent.Parameters
+                    .Where(existingParam => newParameters.All(newParam => newParam.Key != existingParam.Key))
+                    .ToList();
+
+                foreach (var paramToRemove in paramsToRemove)
+                {
+                    existingEvent.Parameters.Remove(paramToRemove);
+                }
+
+                var existingParamsInDb = context.Parameters.ToList();
+                var paramsToAdd = newParameters
+                    .Where(newParam => existingParamsInDb.All(existingParam => existingParam.Key != newParam.Key))
+                    .ToList();
+
+                context.Parameters.AddRange(paramsToAdd);
+                context.SaveChanges();
+
+                var updatedParams = newParameters
+                    .Select(newParam => existingParamsInDb.FirstOrDefault(existingParam => existingParam.Key == newParam.Key) ??
+                                        newParam)
+                    .ToList();
+
+                existingEvent.Parameters = updatedParams;
+
+                context.SaveChanges();
+            }
+            scope.Complete();
+        }
+
+        return eventDto;
+    }
+
+    public bool DeleteEvent(int eventId)
+    {
+        using (var scope = new TransactionScope())
+        {
+            var eventToDelete = context.Events.FirstOrDefault(r => r.Id == eventId);
+
+            if (eventToDelete is null)
+            {
+                return false;
+            }
+

```

Comment:
Do we also want to check for params and tags to delete those (if they are not connected to any other event)